### PR TITLE
Search target element in Rotated Sorted Array 

### PR DESCRIPTION
**Pull request by Bhawana0001**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

YES

### Description

code for Search target element in Rotated Sorted Array added in C++

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Search target element in Rotated Sorted Array in C++ #7105

### CODE REVIEW

**Comment by Bhawana0001** [comment]
@ankitaggarwal23  Sir please Review my PR

**Comment by Bhawana0001** [comment]
@ankitaggarwal23 please review this PR

**Review by siddharth25pandey — APPROVED** [review]
@Bhawana0001 LGTM 💯 Able to merge

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]
@Bhawana0001 Resolve the merge conflicts

**Comment by Bhawana0001** [comment]
@siddharth25pandey  changes done. Sir, Please review it.

File: C-Plus-Plus/cp/Search_target_element_in_Rotated_Sorted_Array.cpp [hunk 1]
```diff
@@ -0,0 +1,108 @@
+
+/*
+Problem:
+Given an array of size n sorted in ascending order is rotated at an unknown pivot index x (0 <= x<n), 
+such that the resulting array is [arr[x], arr[x+1], ..., arr[n-1], arr[0], arr[1], ..., arr[x-1]](0-indexed).
+Given the array arr[] after the rotation and an integer target, return the index of target if it is in array,
+ or -1 if it is not in array.
+
+Write an algorithm with O(log n) runtime complexity.
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/Search_target_element_in_Rotated_Sorted_Array.cpp:9]: remove this line

File: C-Plus-Plus/cp/Search_target_element_in_Rotated_Sorted_Array.cpp [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+*/
+
+/*
+  Explanation:
+     Use Binary search to reduce run time.
+*/
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/Search_target_element_in_Rotated_Sorted_Array.cpp]: remove this redundant comments part

File: C-Plus-Plus/cp/Search_target_element_in_Rotated_Sorted_Array.cpp [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        int i=0;
+        int j=arr.size()-1;
+        
+        if(arr[i]==target)
+                return i;
+            
+            if(arr[j]==target)
+                return j;
+            
+        // Binary Search
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/Search_target_element_in_Rotated_Sorted_Array.cpp]: remove this redundant comment

File: C-Plus-Plus/cp/Search_target_element_in_Rotated_Sorted_Array.cpp [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+int main()
+{
+    int n;  // Number of Element
+    cin>>n;
+
+    vector<int> arr;
+    for(int i=0;i<n;i++)
+    {
+        int x;    // input given array element
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/Search_target_element_in_Rotated_Sorted_Array.cpp]: remove these redundant comments and Use proper display message how you are reading and writing outputs

File: C-Plus-Plus/cp/Search_target_element_in_Rotated_Sorted_Array.cpp [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    vector<int> arr;
+    for(int i=0;i<n;i++)
+    {
+        int x;    // input given array element 
+        cin>>x;
+        arr.push_back(x);
+    }
+   
+     int target;    //input target element
+     cin>>target;
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/Search_target_element_in_Rotated_Sorted_Array.cpp]: remove these redundant comments and Use proper display message how you are reading and writing outputs

File: C-Plus-Plus/cp/Search_target_element_in_Rotated_Sorted_Array.cpp [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+   Example:
+     Input: arr = [4,5,6,7,0,1,2], target = 0
+     Output: 4
+
+     Input: arr=[4,5,6,7,0,1,2], target = 3
+     Output: -1
+
+     Input: arr=[1], target=0
+     Output: -1
+*/
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/Search_target_element_in_Rotated_Sorted_Array.cpp]: Remove this part and rewrite in such format. Also use only one sample case

Comment (ankitaggarwal23) [reply C-Plus-Plus/cp/Search_target_element_in_Rotated_Sorted_Array.cpp]:
```
/*

Sample Input/Output:

Input:

Output:


Time complexity:
Space complexity:

*/
```

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by Bhawana0001** [comment]
@ankitaggarwal23 sir have made the required changes but I when I try to push the changes I don't know what this merge branch thing happened.
please help me to with this.

**Comment by Bhawana0001** [comment]
@ankitaggarwal23 sir have made the required changes but I when I try to push the changes I don't know what this merge branch thing happened.
please help me to with this.

**Review by siddharth25pandey — APPROVED** [review]
@Bhawana0001 LGTM 💯 Able to merge

File: C-Plus-Plus/cp/Search_target_element_in_Rotated_Sorted_Array.cpp [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+such that the resulting array is [arr[x], arr[x+1], ..., arr[n-1], arr[0], arr[1], ..., arr[x-1]](0-indexed).
+Given the array arr[] after the rotation and an integer target, return the index of target if it is in array,
+ or -1 if it is not in array.
+
+Write an algorithm with O(log n) runtime complexity.
+*/
+
+
+#include <iostream>
+#include <vector>
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/Search_target_element_in_Rotated_Sorted_Array.cpp:14]: Replace Header with` #include<bits/stdc++.h>`

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Thank You for the contribution😊

**Pull request merged**